### PR TITLE
DCOS-22310: Update node and npm

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
## Testing

- Run the usual commands, see if everything works as before

## Trade-offs

- I don't have the node and npm version checked in to the package.json because that caused some trouble. I don't see a huge benefit to do it in a backport, so I decided to cut this corner. Feel free to convince me that we need it 😇 
- While trying to fix the bug caused by the checked in node and npm version I tried to update webpack. It did not fix the bug, but seems to work fine otherwise, so I think it makes sense to keep it.

## Dependencies

There is a corresponding plugin branch.

### CODE REVIEW

**Comment by juliangieseke** [comment]
^ reason is that we still have to switch versions otherwise - it might not be as painful as before, but it still adds some overhead 😞

**Comment by DanielMSchmidt** [comment]
@juliangieseke Seems like the error I got when trying to add node and npm as dependencies can be safely ignored. Therefore I add a commit to add them 👍

**Comment by juliangieseke** [comment]
🎉

File: postcss.config.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module.exports = ({ env }) => {
+  const shouldOptimize = env === "production" ? {} : false;
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply postcss.config.js]: It's backported, so let's not change it here. If it is really important to you, let's create a ticket that fixes it on master and 1.11

[the remark this answers is not in the document]
Comment (juliangieseke) [reply postcss.config.js]: no, its not important.

File: package.json [hunk 2]
```diff
@@ -103,57 +103,64 @@
...
+    "webpack-merge": "4.1.1",
+    "webpack-notifier": "1.5.1",
+    "zopfli-webpack-plugin": "0.1.0"
   },
   "scripts": {
     "check-env": "./scripts/validate-engine-versions",
     "clean": "rm -rf dist",
-    "shrinkwrap": "npm shrinkwrap --dev && ./node_modules/.bin/gulp fixShrinkwrap",
+    "build-shrinkwrap": "npm shrinkwrap --dev && ./node_modules/.bin/gulp fixShrinkwrap",
     "preinstall": "./scripts/pre-install",
```
[the remark this answers is not in the document]
Comment (DanielMSchmidt) [reply package.json]: found the root cause, needed to update some random shell script 🤷‍♂️

**Comment by juliangieseke** [comment]
We have to backport our Jenkins setup, am I correct? We had issues building master in dcos with upgraded node version 🤔😕

**Comment by DanielMSchmidt** [comment]
But CI is green, so we should be good, shouldn't we?

**Comment by juliangieseke** [comment]
Yes CI is green and this PR should be good to go, but I'm not sure if we're able to release 1.11 without backporting the tarball-based setup, after we upgraded node 🤔

**Comment by nLight** [comment]
We can merge it and open bumps that will additionally upgrade npm and node.

**Comment by nLight** [comment]
If we want to test dcos build we could clone the repo, make all changes: node/npm versions + dcos-ui sha and try building it

**Comment by nLight** [comment]
@DanielMSchmidt is that you could maybe try?

**Comment by DanielMSchmidt** [comment]
I will first try to do a bump from this branch via CI, afterwards I should be able to make a PR on dcos to see if it works

**Comment by juliangieseke** [comment]
Its green ✅- phew.

**Pull request merged**